### PR TITLE
[azopenai] Adding in support for the latest preview: 2024-03-01

### LINKED ISSUE #22483: Add support for the `dimensions` param of new embedding models
### Feature Request

The `dimensions` param was added to the embeddings request JSON with the new embeddings models, reference [here](https://platform.openai.com/docs/api-reference/embeddings/create#embeddings-create-dimensions).

### PR DESCRIPTION
**Pull request by richardpark-msft**
- Adds in the Dimensions parameter for Embeddings, allowing greater control over the size of the returned embeddings slice.
- Allow controlling the format the embeddings come back as, avoiding potential deserialization errors due to JSON deserialization.

Fixes #22483

### CODE REVIEW

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/ai/azopenai/CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # Release History
 
-## 0.5.1 (Unreleased)
+## 0.6.0 (Unreleased)
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/CHANGELOG.md:3]: Shouldn't this remain `0.5.1`? I don't see any breaking changes.

Comment (richardpark-msft) [reply sdk/ai/azopenai/CHANGELOG.md:3]: Dimensions is a new feature so just going with that. 

Is it not <major>.<minor>.<patch> still, even for betas? Or do you think of it more like 0.<major>.<minor>?

Comment (jhendrixMSFT) [reply sdk/ai/azopenai/CHANGELOG.md:3]: For betas we've been following `0.<major>.<minor>` although I don't think we've been super-stringent on that.

Comment (richardpark-msft) [reply sdk/ai/azopenai/CHANGELOG.md:3]: Ah, okay, didn't realize we had a convention there. I'll bump it back :)

**Review by richardpark-msft — COMMENTED** [review]
/azp run go - azopenai

**Review by richardpark-msft — COMMENTED** [review]
/azp run go - azopenai

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by richardpark-msft** [comment]
(there are some pipeline issues so I'd like to avoid running them again - I'll update the doc comment, as requested in another small PR)

**Pull request merged**